// streaming-availability/dist/models/ChangeType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeTypeToJSON = exports.ChangeTypeFromJSONTyped = exports.ChangeTypeFromJSON = exports.instanceOfChangeType = exports.ChangeType = void 0;
/**
 * Type of the change.
 * @export
 */
exports.ChangeType = {
    New: 'new',
    Removed: 'removed',
    Updated: 'updated',
    Expiring: 'expiring',
    Upcoming: 'upcoming'
};
function instanceOfChangeType(value) {
    return Object.values(exports.ChangeType).includes(value);
}
exports.instanceOfChangeType = instanceOfChangeType;
function ChangeTypeFromJSON(json) {
    return ChangeTypeFromJSONTyped(json, false);
}
exports.ChangeTypeFromJSON = ChangeTypeFromJSON;
function ChangeTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ChangeTypeFromJSONTyped = ChangeTypeFromJSONTyped;
function ChangeTypeToJSON(value) {
    return value;
}
exports.ChangeTypeToJSON = ChangeTypeToJSON;
