// streaming-availability/dist/models/Country.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CountryToJSON = exports.CountryFromJSONTyped = exports.CountryFromJSON = exports.instanceOfCountry = void 0;
var Service_1 = require("./Service");
/**
 * Check if a given object implements the Country interface.
 */
function instanceOfCountry(value) {
    if (!('countryCode' in value))
        return false;
    if (!('name' in value))
        return false;
    if (!('services' in value))
        return false;
    return true;
}
exports.instanceOfCountry = instanceOfCountry;
function CountryFromJSON(json) {
    return CountryFromJSONTyped(json, false);
}
exports.CountryFromJSON = CountryFromJSON;
function CountryFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'countryCode': json['countryCode'],
        'name': json['name'],
        'services': (json['services'].map(Service_1.ServiceFromJSON)),
    };
}
exports.CountryFromJSONTyped = CountryFromJSONTyped;
function CountryToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'countryCode': value['countryCode'],
        'name': value['name'],
        'services': (value['services'].map(Service_1.ServiceToJSON)),
    };
}
exports.CountryToJSON = CountryToJSON;
