// streaming-availability/dist/models/Episode.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.EpisodeToJSON = exports.EpisodeFromJSONTyped = exports.EpisodeFromJSON = exports.instanceOfEpisode = void 0;
/**
 * Check if a given object implements the Episode interface.
 */
function instanceOfEpisode(value) {
    if (!('itemType' in value))
        return false;
    if (!('title' in value))
        return false;
    if (!('airYear' in value))
        return false;
    if (!('streamingOptions' in value))
        return false;
    return true;
}
exports.instanceOfEpisode = instanceOfEpisode;
function EpisodeFromJSON(json) {
    return EpisodeFromJSONTyped(json, false);
}
exports.EpisodeFromJSON = EpisodeFromJSON;
function EpisodeFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'itemType': json['itemType'],
        'title': json['title'],
        'overview': json['overview'] == null ? undefined : json['overview'],
        'airYear': json['airYear'],
        'streamingOptions': json['streamingOptions'],
    };
}
exports.EpisodeFromJSONTyped = EpisodeFromJSONTyped;
function EpisodeToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'itemType': value['itemType'],
        'title': value['title'],
        'overview': value['overview'],
        'airYear': value['airYear'],
        'streamingOptions': value['streamingOptions'],
    };
}
exports.EpisodeToJSON = EpisodeToJSON;
