// streaming-availability/dist/models/Genre.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenreToJSON = exports.GenreFromJSONTyped = exports.GenreFromJSON = exports.instanceOfGenre = void 0;
/**
 * Check if a given object implements the Genre interface.
 */
function instanceOfGenre(value) {
    if (!('id' in value))
        return false;
    if (!('name' in value))
        return false;
    return true;
}
exports.instanceOfGenre = instanceOfGenre;
function GenreFromJSON(json) {
    return GenreFromJSONTyped(json, false);
}
exports.GenreFromJSON = GenreFromJSON;
function GenreFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'name': json['name'],
    };
}
exports.GenreFromJSONTyped = GenreFromJSONTyped;
function GenreToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'name': value['name'],
    };
}
exports.GenreToJSON = GenreToJSON;
