// streaming-availability/dist/models/HorizontalImage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.HorizontalImageToJSON = exports.HorizontalImageFromJSONTyped = exports.HorizontalImageFromJSON = exports.instanceOfHorizontalImage = void 0;
/**
 * Check if a given object implements the HorizontalImage interface.
 */
function instanceOfHorizontalImage(value) {
    if (!('w360' in value))
        return false;
    if (!('w480' in value))
        return false;
    if (!('w720' in value))
        return false;
    if (!('w1080' in value))
        return false;
    if (!('w1440' in value))
        return false;
    return true;
}
exports.instanceOfHorizontalImage = instanceOfHorizontalImage;
function HorizontalImageFromJSON(json) {
    return HorizontalImageFromJSONTyped(json, false);
}
exports.HorizontalImageFromJSON = HorizontalImageFromJSON;
function HorizontalImageFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'w360': json['w360'],
        'w480': json['w480'],
        'w720': json['w720'],
        'w1080': json['w1080'],
        'w1440': json['w1440'],
    };
}
exports.HorizontalImageFromJSONTyped = HorizontalImageFromJSONTyped;
function HorizontalImageToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'w360': value['w360'],
        'w480': value['w480'],
        'w720': value['w720'],
        'w1080': value['w1080'],
        'w1440': value['w1440'],
    };
}
exports.HorizontalImageToJSON = HorizontalImageToJSON;
