// streaming-availability/dist/models/ItemType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ItemTypeToJSON = exports.ItemTypeFromJSONTyped = exports.ItemTypeFromJSON = exports.instanceOfItemType = exports.ItemType = void 0;
/**
 * Type of an item.
 * @export
 */
exports.ItemType = {
    Show: 'show',
    Season: 'season',
    Episode: 'episode'
};
function instanceOfItemType(value) {
    return Object.values(exports.ItemType).includes(value);
}
exports.instanceOfItemType = instanceOfItemType;
function ItemTypeFromJSON(json) {
    return ItemTypeFromJSONTyped(json, false);
}
exports.ItemTypeFromJSON = ItemTypeFromJSON;
function ItemTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ItemTypeFromJSONTyped = ItemTypeFromJSONTyped;
function ItemTypeToJSON(value) {
    return value;
}
exports.ItemTypeToJSON = ItemTypeToJSON;
