// streaming-availability/dist/models/Locale.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.LocaleToJSON = exports.LocaleFromJSONTyped = exports.LocaleFromJSON = exports.instanceOfLocale = void 0;
/**
 * Check if a given object implements the Locale interface.
 */
function instanceOfLocale(value) {
    if (!('language' in value))
        return false;
    return true;
}
exports.instanceOfLocale = instanceOfLocale;
function LocaleFromJSON(json) {
    return LocaleFromJSONTyped(json, false);
}
exports.LocaleFromJSON = LocaleFromJSON;
function LocaleFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'language': json['language'],
        'region': json['region'] == null ? undefined : json['region'],
    };
}
exports.LocaleFromJSONTyped = LocaleFromJSONTyped;
function LocaleToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'language': value['language'],
        'region': value['region'],
    };
}
exports.LocaleToJSON = LocaleToJSON;
