// streaming-availability/dist/models/OrderDirection.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrderDirectionToJSON = exports.OrderDirectionFromJSONTyped = exports.OrderDirectionFromJSON = exports.instanceOfOrderDirection = exports.OrderDirection = void 0;
/**
 *
 * @export
 */
exports.OrderDirection = {
    Asc: 'asc',
    Desc: 'desc'
};
function instanceOfOrderDirection(value) {
    return Object.values(exports.OrderDirection).includes(value);
}
exports.instanceOfOrderDirection = instanceOfOrderDirection;
function OrderDirectionFromJSON(json) {
    return OrderDirectionFromJSONTyped(json, false);
}
exports.OrderDirectionFromJSON = OrderDirectionFromJSON;
function OrderDirectionFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.OrderDirectionFromJSONTyped = OrderDirectionFromJSONTyped;
function OrderDirectionToJSON(value) {
    return value;
}
exports.OrderDirectionToJSON = OrderDirectionToJSON;
