// streaming-availability/dist/models/Price.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PriceToJSON = exports.PriceFromJSONTyped = exports.PriceFromJSON = exports.instanceOfPrice = void 0;
/**
 * Check if a given object implements the Price interface.
 */
function instanceOfPrice(value) {
    if (!('amount' in value))
        return false;
    if (!('currency' in value))
        return false;
    if (!('formatted' in value))
        return false;
    return true;
}
exports.instanceOfPrice = instanceOfPrice;
function PriceFromJSON(json) {
    return PriceFromJSONTyped(json, false);
}
exports.PriceFromJSON = PriceFromJSON;
function PriceFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'amount': json['amount'],
        'currency': json['currency'],
        'formatted': json['formatted'],
    };
}
exports.PriceFromJSONTyped = PriceFromJSONTyped;
function PriceToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'amount': value['amount'],
        'currency': value['currency'],
        'formatted': value['formatted'],
    };
}
exports.PriceToJSON = PriceToJSON;
