// streaming-availability/dist/models/ServiceImageSet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceImageSetToJSON = exports.ServiceImageSetFromJSONTyped = exports.ServiceImageSetFromJSON = exports.instanceOfServiceImageSet = void 0;
/**
 * Check if a given object implements the ServiceImageSet interface.
 */
function instanceOfServiceImageSet(value) {
    if (!('lightThemeImage' in value))
        return false;
    if (!('darkThemeImage' in value))
        return false;
    if (!('whiteImage' in value))
        return false;
    return true;
}
exports.instanceOfServiceImageSet = instanceOfServiceImageSet;
function ServiceImageSetFromJSON(json) {
    return ServiceImageSetFromJSONTyped(json, false);
}
exports.ServiceImageSetFromJSON = ServiceImageSetFromJSON;
function ServiceImageSetFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'lightThemeImage': json['lightThemeImage'],
        'darkThemeImage': json['darkThemeImage'],
        'whiteImage': json['whiteImage'],
    };
}
exports.ServiceImageSetFromJSONTyped = ServiceImageSetFromJSONTyped;
function ServiceImageSetToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'lightThemeImage': value['lightThemeImage'],
        'darkThemeImage': value['darkThemeImage'],
        'whiteImage': value['whiteImage'],
    };
}
exports.ServiceImageSetToJSON = ServiceImageSetToJSON;
