// streaming-availability/dist/models/ServiceInfo.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceInfoToJSON = exports.ServiceInfoFromJSONTyped = exports.ServiceInfoFromJSON = exports.instanceOfServiceInfo = void 0;
var ServiceImageSet_1 = require("./ServiceImageSet");
/**
 * Check if a given object implements the ServiceInfo interface.
 */
function instanceOfServiceInfo(value) {
    if (!('id' in value))
        return false;
    if (!('name' in value))
        return false;
    if (!('homePage' in value))
        return false;
    if (!('themeColorCode' in value))
        return false;
    if (!('imageSet' in value))
        return false;
    return true;
}
exports.instanceOfServiceInfo = instanceOfServiceInfo;
function ServiceInfoFromJSON(json) {
    return ServiceInfoFromJSONTyped(json, false);
}
exports.ServiceInfoFromJSON = ServiceInfoFromJSON;
function ServiceInfoFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'id': json['id'],
        'name': json['name'],
        'homePage': json['homePage'],
        'themeColorCode': json['themeColorCode'],
        'imageSet': (0, ServiceImageSet_1.ServiceImageSetFromJSON)(json['imageSet']),
    };
}
exports.ServiceInfoFromJSONTyped = ServiceInfoFromJSONTyped;
function ServiceInfoToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'id': value['id'],
        'name': value['name'],
        'homePage': value['homePage'],
        'themeColorCode': value['themeColorCode'],
        'imageSet': (0, ServiceImageSet_1.ServiceImageSetToJSON)(value['imageSet']),
    };
}
exports.ServiceInfoToJSON = ServiceInfoToJSON;
