// streaming-availability/dist/models/ShowImageSet.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowImageSetToJSON = exports.ShowImageSetFromJSONTyped = exports.ShowImageSetFromJSON = exports.instanceOfShowImageSet = void 0;
var HorizontalImage_1 = require("./HorizontalImage");
var VerticalImage_1 = require("./VerticalImage");
/**
 * Check if a given object implements the ShowImageSet interface.
 */
function instanceOfShowImageSet(value) {
    if (!('verticalPoster' in value))
        return false;
    if (!('horizontalPoster' in value))
        return false;
    return true;
}
exports.instanceOfShowImageSet = instanceOfShowImageSet;
function ShowImageSetFromJSON(json) {
    return ShowImageSetFromJSONTyped(json, false);
}
exports.ShowImageSetFromJSON = ShowImageSetFromJSON;
function ShowImageSetFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'verticalPoster': (0, VerticalImage_1.VerticalImageFromJSON)(json['verticalPoster']),
        'horizontalPoster': (0, HorizontalImage_1.HorizontalImageFromJSON)(json['horizontalPoster']),
        'verticalBackdrop': json['verticalBackdrop'] == null ? undefined : (0, VerticalImage_1.VerticalImageFromJSON)(json['verticalBackdrop']),
        'horizontalBackdrop': json['horizontalBackdrop'] == null ? undefined : (0, HorizontalImage_1.HorizontalImageFromJSON)(json['horizontalBackdrop']),
    };
}
exports.ShowImageSetFromJSONTyped = ShowImageSetFromJSONTyped;
function ShowImageSetToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'verticalPoster': (0, VerticalImage_1.VerticalImageToJSON)(value['verticalPoster']),
        'horizontalPoster': (0, HorizontalImage_1.HorizontalImageToJSON)(value['horizontalPoster']),
        'verticalBackdrop': (0, VerticalImage_1.VerticalImageToJSON)(value['verticalBackdrop']),
        'horizontalBackdrop': (0, HorizontalImage_1.HorizontalImageToJSON)(value['horizontalBackdrop']),
    };
}
exports.ShowImageSetToJSON = ShowImageSetToJSON;
