// streaming-availability/dist/models/ShowType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowTypeToJSON = exports.ShowTypeFromJSONTyped = exports.ShowTypeFromJSON = exports.instanceOfShowType = exports.ShowType = void 0;
/**
 * Type of a show.
 * @export
 */
exports.ShowType = {
    Movie: 'movie',
    Series: 'series'
};
function instanceOfShowType(value) {
    return Object.values(exports.ShowType).includes(value);
}
exports.instanceOfShowType = instanceOfShowType;
function ShowTypeFromJSON(json) {
    return ShowTypeFromJSONTyped(json, false);
}
exports.ShowTypeFromJSON = ShowTypeFromJSON;
function ShowTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.ShowTypeFromJSONTyped = ShowTypeFromJSONTyped;
function ShowTypeToJSON(value) {
    return value;
}
exports.ShowTypeToJSON = ShowTypeToJSON;
