// streaming-availability/dist/models/StreamingOptionTypes.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingOptionTypesToJSON = exports.StreamingOptionTypesFromJSONTyped = exports.StreamingOptionTypesFromJSON = exports.instanceOfStreamingOptionTypes = void 0;
/**
 * Check if a given object implements the StreamingOptionTypes interface.
 */
function instanceOfStreamingOptionTypes(value) {
    if (!('addon' in value))
        return false;
    if (!('buy' in value))
        return false;
    if (!('rent' in value))
        return false;
    if (!('free' in value))
        return false;
    if (!('subscription' in value))
        return false;
    return true;
}
exports.instanceOfStreamingOptionTypes = instanceOfStreamingOptionTypes;
function StreamingOptionTypesFromJSON(json) {
    return StreamingOptionTypesFromJSONTyped(json, false);
}
exports.StreamingOptionTypesFromJSON = StreamingOptionTypesFromJSON;
function StreamingOptionTypesFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'addon': json['addon'],
        'buy': json['buy'],
        'rent': json['rent'],
        'free': json['free'],
        'subscription': json['subscription'],
    };
}
exports.StreamingOptionTypesFromJSONTyped = StreamingOptionTypesFromJSONTyped;
function StreamingOptionTypesToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'addon': value['addon'],
        'buy': value['buy'],
        'rent': value['rent'],
        'free': value['free'],
        'subscription': value['subscription'],
    };
}
exports.StreamingOptionTypesToJSON = StreamingOptionTypesToJSON;
