// streaming-availability/dist/models/StreamingOptionType.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamingOptionTypeToJSON = exports.StreamingOptionTypeFromJSONTyped = exports.StreamingOptionTypeFromJSON = exports.instanceOfStreamingOptionType = exports.StreamingOptionType = void 0;
/**
 * Type of the streaming option.
 * @export
 */
exports.StreamingOptionType = {
    Free: 'free',
    Subscription: 'subscription',
    Buy: 'buy',
    Rent: 'rent',
    Addon: 'addon'
};
function instanceOfStreamingOptionType(value) {
    return Object.values(exports.StreamingOptionType).includes(value);
}
exports.instanceOfStreamingOptionType = instanceOfStreamingOptionType;
function StreamingOptionTypeFromJSON(json) {
    return StreamingOptionTypeFromJSONTyped(json, false);
}
exports.StreamingOptionTypeFromJSON = StreamingOptionTypeFromJSON;
function StreamingOptionTypeFromJSONTyped(json, ignoreDiscriminator) {
    return json;
}
exports.StreamingOptionTypeFromJSONTyped = StreamingOptionTypeFromJSONTyped;
function StreamingOptionTypeToJSON(value) {
    return value;
}
exports.StreamingOptionTypeToJSON = StreamingOptionTypeToJSON;
