// streaming-availability/dist/models/Subtitle.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubtitleToJSON = exports.SubtitleFromJSONTyped = exports.SubtitleFromJSON = exports.instanceOfSubtitle = void 0;
var Locale_1 = require("./Locale");
/**
 * Check if a given object implements the Subtitle interface.
 */
function instanceOfSubtitle(value) {
    if (!('closedCaptions' in value))
        return false;
    if (!('locale' in value))
        return false;
    return true;
}
exports.instanceOfSubtitle = instanceOfSubtitle;
function SubtitleFromJSON(json) {
    return SubtitleFromJSONTyped(json, false);
}
exports.SubtitleFromJSON = SubtitleFromJSON;
function SubtitleFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'closedCaptions': json['closedCaptions'],
        'locale': (0, Locale_1.LocaleFromJSON)(json['locale']),
    };
}
exports.SubtitleFromJSONTyped = SubtitleFromJSONTyped;
function SubtitleToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'closedCaptions': value['closedCaptions'],
        'locale': (0, Locale_1.LocaleToJSON)(value['locale']),
    };
}
exports.SubtitleToJSON = SubtitleToJSON;
