// streaming-availability/dist/models/VerticalImage.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Streaming Availability API
 * Streaming Availability API allows getting streaming availability information of movies and series; and querying the list of available shows on streaming services such as Netflix, Disney+, Apple TV, Max and Hulu across 60 countries!
 *
 * The version of the OpenAPI document: 4.1.0
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.VerticalImageToJSON = exports.VerticalImageFromJSONTyped = exports.VerticalImageFromJSON = exports.instanceOfVerticalImage = void 0;
/**
 * Check if a given object implements the VerticalImage interface.
 */
function instanceOfVerticalImage(value) {
    if (!('w240' in value))
        return false;
    if (!('w360' in value))
        return false;
    if (!('w480' in value))
        return false;
    if (!('w600' in value))
        return false;
    if (!('w720' in value))
        return false;
    return true;
}
exports.instanceOfVerticalImage = instanceOfVerticalImage;
function VerticalImageFromJSON(json) {
    return VerticalImageFromJSONTyped(json, false);
}
exports.VerticalImageFromJSON = VerticalImageFromJSON;
function VerticalImageFromJSONTyped(json, ignoreDiscriminator) {
    if (json == null) {
        return json;
    }
    return {
        'w240': json['w240'],
        'w360': json['w360'],
        'w480': json['w480'],
        'w600': json['w600'],
        'w720': json['w720'],
    };
}
exports.VerticalImageFromJSONTyped = VerticalImageFromJSONTyped;
function VerticalImageToJSON(value) {
    if (value == null) {
        return value;
    }
    return {
        'w240': value['w240'],
        'w360': value['w360'],
        'w480': value['w480'],
        'w600': value['w600'],
        'w720': value['w720'],
    };
}
exports.VerticalImageToJSON = VerticalImageToJSON;
